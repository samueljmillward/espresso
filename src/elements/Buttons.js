import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ButtonBase from '@material-ui/core/ButtonBase';
import Typography from '@material-ui/core/Typography';

const label = [
  {
    url: '/images/beans.jpg',
    title: 'My Beans',
    width: '50%',
  },
  {
    url: '/images/machine.jpg',
    title: 'Pull A Shot',
    width: '50%',
  },
];

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: 300,
    width: '100%',
    height: '100vh',
  },
  image: {
    position: 'relative',
    height: 200,
    [theme.breakpoints.down('xs')]: {
      width: '100% !important', // Overrides inline-style
      height: 100,
    },
    '&:hover, &$focusVisible': {
      zIndex: 1,
      '& $imageBackdrop': {
        opacity: 0.15,
      },
      '& $imageMarked': {
        opacity: 0,
      },
      '& $imageTitle': {
        border: '4px solid currentColor',
      },
    },
  },
  imageButton: {
    color: theme.palette.common.white,
  },
  imageBackdrop: {
    position: 'absolute',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    backgroundColor: theme.palette.common.black,
    opacity: 0.4,
    transition: theme.transitions.create('opacity'),
  },
  imageTitle: {
    position: 'relative',
    padding: `${theme.spacing(2)}px ${theme.spacing(4)}px ${
      theme.spacing(1) + 6
    }px`,
  },
}));

export default function ButtonBases() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {label.map((image) => (
        <ButtonBase
          focusRipple
          key={image.title}
          className={classes.image}
          style={{
            width: image.width,
          }}
        >
          <span className={classes.imageBackdrop} />
          <span className={classes.imageButton}>
            <Typography
              component="span"
              variant="subtitle1"
              color="inherit"
              className={classes.imageTitle}
            >
              {image.title}
            </Typography>
          </span>
        </ButtonBase>
      ))}
    </div>
  );
}
