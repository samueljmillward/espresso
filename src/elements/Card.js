import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

// import beans from '../images/beans.jpg';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
}));

export default function BeanCard() {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Typography
          className={classes.title}
          color="textSecondary"
          gutterBottom
        >
          Roasted on: 18th November.
        </Typography>
        <Typography variant="h5" component="h2">
          House Espresso
        </Typography>
        <Typography className={classes.pos} color="textSecondary">
          Ethiopia
          <br />
          caramel, dark chocolate, toffee
        </Typography>
      </CardContent>
      {/* <CardMedia
        component="img"
        style={{ height: '300px' }}
        src={beans}
        title="House Espresso"
      /> */}
      <CardActions>
        <IconButton aria-label="add to favorites">
          <FavoriteIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body2" component="p">
            Grind: 6
          </Typography>
          <Typography className={classes.pos} variant="body2" component="p">
            <br />
            Recipe
            <br />
            Dry: 18g
            <br />
            Final Weight: 36g
          </Typography>
          <Typography variant="body2" component="p">
            Our house espresso is a modern take on a classic. It’s a blend of
            two distinctive coffees, which combine to produce an espresso with a
            deep, rich chocolatey body, pronounced toffee sweetness, a complex
            fruity edge, and a hint of bittersweet cocoa at the finish. It’s
            well balanced and combines beautifully with milk, where it adds
            punchy caramel, smooth milk chocolate notes and gets even sweeter
          </Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
