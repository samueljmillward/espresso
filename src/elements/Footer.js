import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
    footer: {
        color: '#c46404',
        backgroundColor: '#642b09',
        paddingTop: '3em',
        width: '100%',
        position: 'absolute',
        bottom: 0,
    },
    col: {
        display: 'flex',
        justifyContent: 'center',
        flexDirection: 'row',
        alignItems: 'center',
        margin: 0,
        padding: '.2rem',
    },
}));

const Footer = () => {
    const classes = useStyles();

    return (
        <div className={classes.footer}>
            <div>
                <p className={classes.col}>&copy;{new Date().getFullYear()} Espresso</p>
            </div>
        </div>
    );
};

export default Footer;
