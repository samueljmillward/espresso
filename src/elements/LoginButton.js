import React from 'react';
import styled from 'styled-components';
import { useAuth0 } from '@auth0/auth0-react';

const LoginButton = () => {
  const { loginWithRedirect } = useAuth0();
  return <LoginBtn onClick={() => loginWithRedirect()}>Login</LoginBtn>;
};

const LoginBtn = styled.a`
  padding: 1rem 2rem;
  cursor: pointer;
  text-align: center;
  text-decoration: none;
  color: burlywood;
  transition: all 0.3s ease-in;
  font-size: 0.9rem;

  &:hover {
    color: #c46404;
  }
`;

export default LoginButton;
