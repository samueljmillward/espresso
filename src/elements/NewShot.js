import React from 'react';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
// import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import DateFnsUtils from '@date-io/date-fns';
import {
  MuiPickersUtilsProvider,
  KeyboardDatePicker,
} from '@material-ui/pickers';
import { useForm, Controller } from 'react-hook-form';

import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  padding: {
    paddingBottom: '1.2rem',
  },
  weightPickers: {
    display: 'flex',
  },
  addButton: {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: '32px',
  },
}));

export default function NewShot() {
  const classes = useStyles();

  const [open, setOpen] = React.useState(true);

  const { register, handleSubmit, control } = useForm();

  const onSubmit = (data) => {
    handleClose();
    console.clear();
    console.log({ data });
  };

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const [selectedDate, setSelectedDate] = React.useState(
    new Date('2020-11-22T21:11:54')
  );

  const [values, setValues] = React.useState({
    weight: '',
  });

  const handleDateChange = (date) => {
    setSelectedDate(date);
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  return (
    <div className={classes.addButton}>
      <Button variant="contained" color="primary" onClick={handleClickOpen}>
        New Shot
      </Button>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">New Espresso Shot</DialogTitle>
        <form onSubmit={handleSubmit(onSubmit)}>
          <DialogContent>
            {/* <DialogContentText>
            To subscribe to this website, please enter your email address here.
            We will send updates occasionally.
          </DialogContentText> */}
            <TextField
              className={classes.padding}
              autoFocus
              margin="dense"
              inputRef={register}
              name="name"
              id="name"
              label="Name"
              type="text"
              fullWidth
            />
            <div className={classes.weightPickers}>
              <div className={classes.padding}>
                <TextField
                  id="outlined-adornment-weight"
                  onChange={handleChange('weight')}
                  name="weight"
                  control={control}
                  defaultValue="0"
                  aria-describedby="outlined-weight-helper-text"
                  inputProps={{
                    'aria-label': 'weight',
                  }}
                  label="Dry Weight (g)"
                  type="number"
                  InputLabelProps={{
                    shrink: true,
                  }}
                  variant="outlined"
                />
              </div>
              <TextField
                id="standard-number"
                label="Grind"
                type="number"
                variant="outlined"
                InputLabelProps={{
                  shrink: true,
                }}
              />
            </div>
            <div className={classes.finalWeight}>
              <TextField
                id="outlined-adornment-weight"
                onChange={handleChange('weight')}
                name="weight"
                control={control}
                defaultValue="0"
                aria-describedby="outlined-weight-helper-text"
                inputProps={{
                  'aria-label': 'weight',
                }}
                label="Final Weight (g)"
                type="number"
                InputLabelProps={{
                  shrink: true,
                }}
                variant="outlined"
              />
            </div>
            <MuiPickersUtilsProvider utils={DateFnsUtils}>
              <Grid container justify="space-around">
                <Controller
                  as={<KeyboardDatePicker />}
                  disableToolbar
                  variant="inline"
                  format="MM/dd/yyyy"
                  margin="normal"
                  name="roastDate"
                  control={control}
                  defaultValue="2020-11-28"
                  id="date-picker-inline"
                  label="Date picker inline"
                  value={selectedDate}
                  onChange={handleDateChange}
                  fullWidth
                  KeyboardButtonProps={{
                    'aria-label': 'change date',
                  }}
                />
              </Grid>
            </MuiPickersUtilsProvider>
            <DialogActions>
              <Button onClick={handleClose} color="primary">
                Cancel
              </Button>
              <Button type="submit" color="primary">
                Submit
              </Button>
            </DialogActions>
          </DialogContent>
        </form>
      </Dialog>
    </div>
  );
}
