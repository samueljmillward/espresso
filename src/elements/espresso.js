// eslint-disable-next-line no-lone-blocks
{
  /* <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body2" component="p">
            Grind: 6
          </Typography>
          <Typography className={classes.pos} variant="body2" component="p">
            <br />
            Recipe
            <br />
            Dry: 18g
            <br />
            Final Weight: 36g
          </Typography>
          <Typography variant="body2" component="p">
            Our house espresso is a modern take on a classic. It’s a blend of
            two distinctive coffees, which combine to produce an espresso with a
            deep, rich chocolatey body, pronounced toffee sweetness, a complex
            fruity edge, and a hint of bittersweet cocoa at the finish. It’s
            well balanced and combines beautifully with milk, where it adds
            punchy caramel, smooth milk chocolate notes and gets even sweeter
          </Typography>
        </CardContent>
      </Collapse> */
}
