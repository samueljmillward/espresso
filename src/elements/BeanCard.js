import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Container from '@material-ui/core/Container';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import FavoriteIcon from '@material-ui/icons/Favorite';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  container: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
  },
  paddingTop: {
    paddingTop: 0,
  },
}));

export default function BeanCard({ bean }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant='outlined'>
      <CardContent>
        <Typography
          className={classes.title}
          color='textSecondary'
          gutterBottom
        >
          Roast Date: {bean.roastDate}
        </Typography>
        <Typography variant='h5' component='h2'>
          {bean.name}
        </Typography>
        <Typography className={classes.pos} color='textSecondary'>
          {bean.origin}
        </Typography>
        <Container className={classes.container}>
          <Typography color='textSecondary'>{bean.flavours}</Typography>
          <Typography color='textSecondary'>{bean.weight}g</Typography>
        </Container>
      </CardContent>
      <CardActions className={classes.paddingTop}>
        <IconButton aria-label='add to favorites'>
          <FavoriteIcon />
        </IconButton>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label='show more'
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout='auto' unmountOnExit>
        <CardContent>
          <Typography variant='body1'>Notes:</Typography>
          <Typography variant='body2'>{bean.notes}</Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
