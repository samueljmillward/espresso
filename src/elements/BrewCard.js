import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
// import CardMedia from '@material-ui/core/CardMedia';
import CardActions from '@material-ui/core/CardActions';
import Container from '@material-ui/core/Container';
import Collapse from '@material-ui/core/Collapse';
import IconButton from '@material-ui/core/IconButton';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

// import beans from '../images/beans.jpg';

const useStyles = makeStyles((theme) => ({
  root: {
    minWidth: 275,
  },
  date: {
    display: 'flex',
    justifyContent: 'flex-end',
  },
  expand: {
    transform: 'rotate(0deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: 'rotate(180deg)',
  },
  container: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
  },
  paddingBottom: {
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'space-between',
    paddingBottom: '.8rem',
  },
}));

export default function BrewCard({ brews }) {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  return (
    <Card className={classes.root} variant="outlined">
      <CardContent>
        <Container className={classes.container}>
          <Typography variant="h5" component="h2">
            {brews.name}
          </Typography>
          <Typography className={classes.date} color="textSecondary">
            Brewed on: {brews.brewDate}
          </Typography>
        </Container>
        <Typography className={classes.paddingBottom} variant="caption">
          Espresso
        </Typography>
        <Container className={classes.paddingBottom}>
          <Typography variant="h6" color="textSecondary">
            Dry Weight: {brews.dryWeight}g
          </Typography>
          <Typography variant="h6" color="textSecondary">
            Grind: {brews.grind}
          </Typography>
        </Container>
        <Container className={classes.container}>
          <Typography variant="h6" color="textSecondary">
            Final Weight: {brews.finalWeight}g
          </Typography>
          <Typography variant="h6" color="textSecondary">
            Ratio: 1:1
          </Typography>
        </Container>
      </CardContent>
      {/* <CardMedia
        component="img"
        style={{ height: '300px' }}
        src={beans}
        title="House Espresso"
      /> */}
      <CardActions className={classes.paddingTop}>
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded,
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
        >
          <ExpandMoreIcon />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
        <CardContent>
          <Typography variant="body1">Notes:</Typography>
          <Typography variant="body1">{brews.notes}</Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
}
