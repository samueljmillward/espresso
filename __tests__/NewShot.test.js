import { defaultValues } from '../src/elements/NewShot';

describe('returns expected data types', () => {
  test('defaultValues match schema', () => {
    const shotSchema = {
      name: expect.any(String),
      dryWeight: expect.any(Number),
      grind: expect.any(Number),
      weight: expect.any(Number),
      notes: expect.any(String),
      brewDate: expect.any(Object),
    };
    expect(defaultValues).toEqual(shotSchema);
  });
});
