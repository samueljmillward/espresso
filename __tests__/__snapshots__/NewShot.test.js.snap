// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI NewShot form 1`] = `
<div
  className="makeStyles-buttons-2"
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-addButton-4"
    onClick={[Function]}
    variant="contained"
  >
    New Shot
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-deleteButton-5"
    onClick={[Function]}
    variant="contained"
  >
    Delete Brews
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Dialog))
    aria-labelledby="form-dialog-title"
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(DialogTitle))
      id="form-dialog-title"
    >
      New Espresso Shot
    </WithStyles(ForwardRef(DialogTitle))>
    <form
      onSubmit={[Function]}
    >
      <WithStyles(ForwardRef(DialogContent))>
        <WithStyles(ForwardRef(TextField))
          autoFocus={true}
          defaultValue="Old Brown Java"
          fullWidth={true}
          id="name"
          inputRef={[Function]}
          label="Name"
          margin="dense"
          name="name"
          type="text"
        />
        <p
          className="makeStyles-errorMessage-3"
        />
        <div
          className="makeStyles-weightPickers-1"
        >
          <div>
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              aria-describedby="outlined-weight-helper-text"
              defaultValue={18}
              id="standard-number"
              inputRef={[Function]}
              label="Dry Weight (g)"
              name="dryWeight"
              pattern="^-?[0-9]\\\\d*\\\\.?\\\\d*$"
              type="number"
              variant="outlined"
            />
            <p
              className="makeStyles-errorMessage-3"
            />
          </div>
          <WithStyles(ForwardRef(TextField))
            InputLabelProps={
              Object {
                "shrink": true,
              }
            }
            defaultValue={3}
            id="standard-number"
            inputRef={[Function]}
            label="Grind"
            name="grind"
            pattern="^-?[0-9]\\\\d*\\\\.?\\\\d*$"
            type="number"
            variant="outlined"
          />
          <p
            className="makeStyles-errorMessage-3"
          />
        </div>
        <div>
          <WithStyles(ForwardRef(FormHelperText))
            id="outlined-weight-helper-text"
          >
            Final Weight (g)
          </WithStyles(ForwardRef(FormHelperText))>
          <Controller
            aria-describedby="outlined-weight-helper-text"
            as={<WithStyles(ForwardRef(OutlinedInput)) />}
            control={
              Object {
                "defaultValuesRef": Object {
                  "current": Object {},
                },
                "fieldArrayDefaultValuesRef": Object {
                  "current": Object {},
                },
                "fieldArrayNamesRef": Object {
                  "current": Set {},
                },
                "fieldArrayValuesRef": Object {
                  "current": Object {},
                },
                "fieldsRef": Object {
                  "current": Object {},
                },
                "fieldsWithValidationRef": Object {
                  "current": Object {},
                },
                "formState": Object {
                  "dirtyFields": Object {},
                  "errors": Object {},
                  "isDirty": false,
                  "isSubmitSuccessful": false,
                  "isSubmitted": false,
                  "isSubmitting": false,
                  "isValid": false,
                  "isValidating": false,
                  "submitCount": 0,
                  "touched": Object {},
                },
                "formStateRef": Object {
                  "current": Object {
                    "dirtyFields": Object {},
                    "errors": Object {},
                    "isDirty": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "submitCount": 0,
                    "touched": Object {},
                  },
                },
                "getValues": [Function],
                "isFormDirty": [Function],
                "mode": Object {
                  "isOnAll": false,
                  "isOnBlur": false,
                  "isOnChange": false,
                  "isOnSubmit": true,
                  "isOnTouch": false,
                },
                "reValidateMode": Object {
                  "isReValidateOnBlur": false,
                  "isReValidateOnChange": true,
                },
                "readFormStateRef": Object {
                  "current": Object {
                    "constructor": true,
                    "dirtyFields": true,
                    "errors": true,
                    "isDirty": true,
                    "isSubmitSuccessful": true,
                    "isSubmitted": true,
                    "isSubmitting": true,
                    "isValid": true,
                    "isValidating": true,
                    "submitCount": true,
                    "touched": true,
                  },
                },
                "register": [Function],
                "removeFieldEventListener": [Function],
                "resetFieldArrayFunctionRef": Object {
                  "current": Object {},
                },
                "setValue": [Function],
                "shallowFieldsStateRef": Object {
                  "current": Object {},
                },
                "shouldUnregister": true,
                "trigger": [Function],
                "unregister": [Function],
                "updateFormState": [Function],
                "updateWatchedValue": [Function],
                "useWatchFieldsRef": Object {
                  "current": Object {},
                },
                "useWatchRenderFunctionsRef": Object {
                  "current": Object {},
                },
                "validFieldsRef": Object {
                  "current": Object {},
                },
                "validateResolver": [Function],
                "watchInternal": [Function],
              }
            }
            defaultValue={36}
            id="outlined-adornment-weight"
            inputProps={
              Object {
                "aria-label": "weight",
              }
            }
            labelWidth={0}
            name="weight"
            onChange={[Function]}
            pattern="^-?[0-9]\\\\d*\\\\.?\\\\d*$"
            rules={
              Object {
                "max": 80,
                "maxLength": 2,
                "min": 1,
                "required": true,
              }
            }
            type="number"
          />
          <p
            className="makeStyles-errorMessage-3"
          />
        </div>
        <WithStyles(ForwardRef(TextField))
          autoFocus={true}
          defaultValue="Slightly bitter"
          fullWidth={true}
          id="notes"
          inputRef={[Function]}
          label="Notes"
          margin="dense"
          name="notes"
          type="text"
        />
        <p
          className="makeStyles-errorMessage-3"
        />
        <MuiPickersUtilsProvider
          utils={[Function]}
        >
          <WithStyles(ForwardRef(Grid))
            container={true}
            justify="space-around"
          >
            <Controller
              KeyboardButtonProps={
                Object {
                  "aria-label": "change date",
                }
              }
              as={
                <PickerWithState
                  allowKeyboardControl={true}
                  invalidDateMessage="Invalid Date Format"
                  maxDate={2100-01-01T00:00:00.000Z}
                  maxDateMessage="Date should not be after maximal date"
                  minDate={1900-01-01T00:00:00.000Z}
                  minDateMessage="Date should not be before minimal date"
                  openTo="date"
                  views={
                    Array [
                      "year",
                      "date",
                    ]
                  }
                />
              }
              control={
                Object {
                  "defaultValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldArrayDefaultValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldArrayNamesRef": Object {
                    "current": Set {},
                  },
                  "fieldArrayValuesRef": Object {
                    "current": Object {},
                  },
                  "fieldsRef": Object {
                    "current": Object {},
                  },
                  "fieldsWithValidationRef": Object {
                    "current": Object {},
                  },
                  "formState": Object {
                    "dirtyFields": Object {},
                    "errors": Object {},
                    "isDirty": false,
                    "isSubmitSuccessful": false,
                    "isSubmitted": false,
                    "isSubmitting": false,
                    "isValid": false,
                    "isValidating": false,
                    "submitCount": 0,
                    "touched": Object {},
                  },
                  "formStateRef": Object {
                    "current": Object {
                      "dirtyFields": Object {},
                      "errors": Object {},
                      "isDirty": false,
                      "isSubmitSuccessful": false,
                      "isSubmitted": false,
                      "isSubmitting": false,
                      "isValid": false,
                      "isValidating": false,
                      "submitCount": 0,
                      "touched": Object {},
                    },
                  },
                  "getValues": [Function],
                  "isFormDirty": [Function],
                  "mode": Object {
                    "isOnAll": false,
                    "isOnBlur": false,
                    "isOnChange": false,
                    "isOnSubmit": true,
                    "isOnTouch": false,
                  },
                  "reValidateMode": Object {
                    "isReValidateOnBlur": false,
                    "isReValidateOnChange": true,
                  },
                  "readFormStateRef": Object {
                    "current": Object {
                      "constructor": true,
                      "dirtyFields": true,
                      "errors": true,
                      "isDirty": true,
                      "isSubmitSuccessful": true,
                      "isSubmitted": true,
                      "isSubmitting": true,
                      "isValid": true,
                      "isValidating": true,
                      "submitCount": true,
                      "touched": true,
                    },
                  },
                  "register": [Function],
                  "removeFieldEventListener": [Function],
                  "resetFieldArrayFunctionRef": Object {
                    "current": Object {},
                  },
                  "setValue": [Function],
                  "shallowFieldsStateRef": Object {
                    "current": Object {},
                  },
                  "shouldUnregister": true,
                  "trigger": [Function],
                  "unregister": [Function],
                  "updateFormState": [Function],
                  "updateWatchedValue": [Function],
                  "useWatchFieldsRef": Object {
                    "current": Object {},
                  },
                  "useWatchRenderFunctionsRef": Object {
                    "current": Object {},
                  },
                  "validFieldsRef": Object {
                    "current": Object {},
                  },
                  "validateResolver": [Function],
                  "watchInternal": [Function],
                }
              }
              defaultValue={2021-05-15T00:00:00.000Z}
              disableToolbar={true}
              format="dd/MM/yyyy"
              fullWidth={true}
              id="date-picker-inline"
              label="Brewed on: "
              margin="normal"
              name="brewDate"
              onChange={[Function]}
              value={2021-05-15T00:00:00.000Z}
              variant="inline"
            />
          </WithStyles(ForwardRef(Grid))>
        </MuiPickersUtilsProvider>
        <WithStyles(ForwardRef(DialogActions))>
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[Function]}
          >
            Cancel
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            color="primary"
            type="submit"
          >
            Submit
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(DialogActions))>
      </WithStyles(ForwardRef(DialogContent))>
    </form>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
